engine/src/main/java/org/destinationsol/game/Console.java

```diff
@@ -0,0 +1,198 @@
+        }
+    }
+
+    @Override
+    public boolean isCursorOnBackground(SolInputManager.InputPointer inputPointer) {
+        return true;
+    }
+
+    @Override
+    public void blurCustom(SolApplication solApplication) {
+        isActive = false;
+    }
+
+    @Override
+    public void onAdd(SolApplication solApplication) {
+        isActive = true;
+    }
+
+    @Override
+    public List<SolUiControl> getControls() {
+        return controls;
+    }
+
+    @Override
+    public void drawBackground(UiDrawer uiDrawer, SolApplication solApplication) {
+        drawFrame(uiDrawer);
+        // Text area
+        uiDrawer.draw(new Rectangle(TOP_LEFT.x + 2 * FRAME_WIDTH,
+                        TOP_LEFT.y + 2 * FRAME_WIDTH,
+                        (BOTTOM_RIGHT.x - TOP_LEFT.x) - 4 * FRAME_WIDTH,
+                        BOTTOM_RIGHT.y - TOP_LEFT.y - 4 * FRAME_WIDTH),
+                SolColor.UI_BG_LIGHT);
+        drawTextEntrySeparator(uiDrawer);
+    }
+
+    private void drawTextEntrySeparator(UiDrawer uiDrawer) {
+        uiDrawer.drawLine(TOP_LEFT.x + 2 * FRAME_WIDTH, getLineY(20.333f), 0, (BOTTOM_RIGHT.x - TOP_LEFT.x) - 4 * FRAME_WIDTH, Color.WHITE);
+    }
+
+    private void drawFrame(UiDrawer uiDrawer) {
+        uiDrawer.draw(new Rectangle(TOP_LEFT.x, TOP_LEFT.y,
+                        (BOTTOM_RIGHT.x - TOP_LEFT.x), BOTTOM_RIGHT.y - TOP_LEFT.y),
+                SolColor.UI_LIGHT);
+        uiDrawer.draw(new Rectangle(TOP_LEFT.x + FRAME_WIDTH, TOP_LEFT.y + FRAME_WIDTH,
+                        (BOTTOM_RIGHT.x - TOP_LEFT.x) - 2 * FRAME_WIDTH, BOTTOM_RIGHT.y - TOP_LEFT.y - 2 * FRAME_WIDTH),
+                SolColor.UI_BG_LIGHT);
+    }
+
+    @Override
+    public void drawText(UiDrawer uiDrawer, SolApplication solApplication) {
+        String s = "randomtext";
+        final float textX = TOP_LEFT.x + 2 * FRAME_WIDTH;
+        for (int line = 0; line < 20; line++) {
+            if (linesOfOutput.size() + line > 19) {
+                final String text = linesOfOutput.get(linesOfOutput.size() - 20 + line);
+                uiDrawer.drawString(text, textX, getLineY(line), 0.5f, UiDrawer.TextAlignment.LEFT, false, Color.WHITE);
+            }
+        }
+        drawInputLine(uiDrawer, textX);
+
+    }
+
+    private void drawInputLine(UiDrawer uiDrawer, float textX) {
+        StringBuilder stringBuilder = new StringBuilder();
+        int width = 0;
+        for (char c : inputLine.reverse().toString().toCharArray()) {
+            try {
+                width += (c == ' ' ? 3 : 1) * font.getData().getGlyph(c).width;
+                if (width > MAX_WIDTH_OF_LINE) {
+                    break;
+                }
+                stringBuilder.append(c);
+            } catch (NullPointerException e) {
+                inputLine.deleteCharAt(0); // when there is character that can't be handled by font, it'll always be the last (first in reversed), since if it was anywhere elsewhere, it'd already be removed earlier.
```

Comment:
This sort of exception handling looks like a poor practice (exception handling of runtime issues should not be part of the control flow or something). How about detecting the nulls and reacting to them in the method itself instead? Then hopefully there'll be a good place to put better comments there rather than as a sort of dangling long comment in a catch block that's easy to ignore

Reply:
Bump - any resolution on this? :-)